docker-compose-rule-core/src/main/java/com/palantir/docker/compose/reporting/HttpJsonPoster.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.docker.compose.reporting;
+
+import com.google.common.io.CharStreams;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class HttpJsonPoster {
+    private static final Logger log = LoggerFactory.getLogger(HttpJsonPoster.class);
+    private static final int TIMEOUT = 10_000;
+    private static final String REPORT_API_VERSION = "2";
+
+    private final ReportingConfig reportingConfig;
+
+    HttpJsonPoster(ReportingConfig reportingConfig) {
+        this.reportingConfig = reportingConfig;
+    }
+
+    public void post(String json) {
+        try {
+            URL url = new URL(templateVersion(reportingConfig.url()));
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            connection.setConnectTimeout(TIMEOUT);
+            connection.setReadTimeout(TIMEOUT);
+
+            connection.setRequestMethod("POST");
+            connection.setInstanceFollowRedirects(true);
+            connection.setRequestProperty("Content-Type", "application/json");
+
+            String version = Optional.ofNullable(this.getClass().getPackage().getImplementationVersion())
+                    .orElse("0.0.0");
+            connection.setRequestProperty("User-Agent", "docker-compose-rule/" + version);
+
+            connection.setDoOutput(true);
+            PrintWriter body = new PrintWriter(connection.getOutputStream());
+            body.println(json);
+            body.close();
+
+            connection.connect();
```

Comment:
Long term, we could probably buffer these up somewhere in the home dir if we wanted to resume POSTing them for aeroplane dev